src/engine/enginebuffer.h

```diff
@@ -346,11 +347,12 @@ class EngineBuffer : public EngineObject {
 
     // Object used to perform waveform scaling (sample rate conversion)
     EngineBufferScale* m_pScale;
+    FRIEND_TEST(EngineBufferTest, SlowRubberBand);
     // Object used for linear interpolation scaling of the audio
-    EngineBufferScaleLinear* m_pScaleLinear;
+    EngineBufferScale* m_pScaleLinear;
     // Object used for pitch-indep time stretch (key lock) scaling of the audio
-    EngineBufferScaleST* m_pScaleST;
-    EngineBufferScaleRubberBand* m_pScaleRB;
+    EngineBufferScale* m_pScaleST;
+    EngineBufferScale* m_pScaleRB;
```

Comment:
The idea was to have two pointers to the linear scaler. On with it natural type and one that is uses during process(). We have already m_pScaleKeylock for the same purpose.